module-2/backend/week_one.md

```diff
@@ -4,29 +4,104 @@ Fork this respository. Answer the questions to the best of your ability. Try to
-13. What does HTTP stand for? 
-14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
-15. What's an ORM?
-16. What's the most commonly used ORM?
-17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
-19. When you create a migration, does it automatically modify your database?
-20. How does a model relate to a database?
+  
+```ruby
+get '/horses' do
+  name = "Mr. Ed"
+  erb :index, locals: {name: name}
+end
+```
+9. What's the purpose of ERB?<br>
+
+Erb allows us to inject the results of our ruby code into some other type of code (i.e. html). I think of it like interpolation but in the context of a file instead of a string. This allows us to pull data from our model and wrap it into html which is appropriate for presentation in the browser.
+
+10. Why do I need a development AND test database?<br>
+
+You need both databases because you don't want your tests to be manipulating the actual dataset you're working with. I think I heard something like dirtying/muddling the original database. Additionally in our testing environment we're "cleaning" our database between each "it" block.
+
+11. What's responsive design?<br>
+
+Responsive design refers to web pages that can be displayed beautifully regardless of the device the viewer is using.
+
+12. What is CRUD and why is it important?<br>
+
+CRUD is a design paradigm with databases for persistent storage. These are all the functions you need to have a functional data storage system. Create, Read, Update, and Delete.
+
+13. What does HTTP stand for? <br>
+
+HyperText Transfer Protocol
+
+14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?<br>
+
+`<%= %>` and `<% %>`. The first displays the evaluation of the ruby code within while the second evaluates but doesn't display.
+
+15. What's an ORM?<br>
+
+An ORM is an object relational model. This describes the situation where an object is a go between a database and the view in MVC.
+
+16. What's the most commonly used ORM?<br>
+
+ActiveRecord 
+
+17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for. <br>
+
+get '/restaurants' to show all restaurants
+
+get '/restaurants/:id' to show an individual restaurant
+
+get '/restaurants/new' to show the form for adding a new restaurant
+
+post '/restaurants/new' to actually add the stuff from the form, touching the database
+
+get '/restaurants/:id/edit' to show the form for editing a restuarant
+
+post '/restaurants/:id' to actually add the from from the form, updating the info in the database

```

Comment:
this will be a `PUT` rather than a `POST`. post is for creating new rows in our database and put/patch are for updating a specific row